src/invariant/ConservationOfLumens.cpp

```diff
@@ -71,7 +79,54 @@ calculateDeltaBalance(LedgerEntry const* current, LedgerEntry const* previous)
     }
 #ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
     case CONTRACT_DATA:
-        break;
+    {
+        auto const& contractData = current ? current->data.contractData()
+                                           : previous->data.contractData();
+        if (contractData.contractID != lumenContractID ||
+            contractData.key.type() != SCV_VEC || !contractData.key.vec() ||
+            contractData.key.vec().size() == 0)
+        {
+            return 0;
+        }
+
+        // The balanceSymbol should be the first entry in the SCVec
+        if (!(contractData.key.vec()->at(0) == balanceSymbol))
+        {
+            return 0;
+        }
+
+        auto getAmount = [&amountSymbol](LedgerEntry const* entry) -> int64_t {
+            if (!entry)
+            {
+                return 0;
+            }
+
+            // The amount should be the first entry in the SCMap
+            auto const& val = entry->data.contractData().val;
+            if (val.type() == SCV_MAP && val.map() && val.map()->size() != 0)
+            {
+                auto const& amountEntry = val.map()->at(0);
+                if (amountEntry.key == amountSymbol)
+                {
+                    if (amountEntry.val.type() == SCV_I128)
+                    {
+                        auto lo = amountEntry.val.i128().lo;
+                        auto hi = amountEntry.val.i128().hi;
+                        if (lo > INT64_MAX || hi > 0)
+                        {
+                            // The amount isn't right, but it'll trigger the
+                            // invariant.
+                            return INT64_MAX;
```

Comment:
those conditions fall under "malformed" and we definitely would want to catch this all the time (but the actual failure belongs to a different invariant).
I don't think the code as written will always trigger an invariant failure (so the comment is a bit misleading)

Reply:
Yeah this can be included in the other invariant. I'll update the comment. I don't think it's worth changing the logic here though to use a int128. I could update this method to return a string as well that could contain an error, but that error would be covered by the other invariant.